### ISSUE TITLE
Screen Reader: Cannot Add a Widget on Mobile Web

### ISSUE THREAD

**Issue by donalirl** [Automattic/wp-calypso#43504]
When adding a new widget on mobile web, the list of widgets pops up in some kind of modal window. The problem is that this modal cannot be read by screen readers, so visually-impaired users cannot interact with it.

![Screen_Recording_20200621-133726_1](https://user-images.githubusercontent.com/30727666/85216243-7faa0700-b3c5-11ea-9fed-cfc2d159a2a1.gif)

#### Steps to reproduce

1. Starting at My Site → Design → Customize → Widgets, choose any widget area
2. Activate a screen reader on your device
3. Navigate to the Add a Widget button and tap it
4. Attempt to continue with the task of adding a widget, but the screen reader cannot find it.

#### What I expected

I expected to be able to add a widget using a screen reader.

#### What happened instead

A visually-impaired user on a screen reader cannot add a widget. The screen goes "blank" to them after tapping Add a Widget

#### Browser / OS version

Device: iPod Touch
Browser: Safari Latest
Screen reader: Apple Voiceover

Device: Samsung Galaxy S10e, Android 10
Browser: Chrome 83
Screen reader: Android Voice Assistant

#### Context / Source

#3030290-zen

**Comment by jeryj** [comment]
I've tested on my iPhone using VoiceOver and can confirm. You _can_ add a widget by dragging your finger around the screen to locate the search field and widget options, but using the swiping will never announce the widgets. I also noticed that the customizer in general doesn't seem to be able to be accessible using a keyboard or swiping. Focus is never moved into the customizer after clicking the "Customizer" link. Focus remains on the Customizer link in the menu with the customizer loading as an overlay on top of it.

My recommendation (we'll see what's possible though!) is:
1. When the customizer link is pressed, move focus into the customizer. This is a new page, so there should be no concept of the previous page (using the back button on the browser routes you to the previous page).
2. On mobile devices, when you press the Add Widget button, focus should be moved to the widget search input. This does work on desktop with Chrome and Safari. I'm not sure why Safari iOS doesn't do this. On my testing focus returned to the document root (which is the `/home` page behind the customizer), which suggests a focus loss.

I'll keep debugging and open a PR if I can find a solution.

**Comment by jeryj** [comment]
> When the customizer link is pressed, move focus into the customizer. This is a new page, so there should be no concept of the previous page (using the back button on the browser routes you to the previous page).

I have a [PR that moves focus into the Customizer iframe](https://github.com/Automattic/wp-calypso/pull/44573) after you click on Customize, but it doesn't trap focus/block access to the page behind it. Calypso is loading an iframe from core, so I think we could at least trap focus on the customizer, but changes within the customizer would need to be done in core.

> On mobile devices, when you press the Add Widget button, focus should be moved to the widget search input.

This is a WordPress.org Core issue. I opened a ticket for it with the potential solution, but I don't know if it will be accepted: https://core.trac.wordpress.org/ticket/50846#ticket

Even though focus is not moved, I believe many VoiceOver users (I am by no means an expert so I could be very, very wrong here!) use their finger to announce what's on the screen by moving around rather than swiping from one item to the next, such as [this person demonstrates on Twitter](https://twitter.com/Kristy_Viers/status/1287189581926981634).

cc @davemart-in to see how you'd like to proceed with this task

**Comment by jeryj** [comment]
https://github.com/Automattic/wp-calypso/pull/44573 is merged, which improves the focus management so you're:
1.  Placed in the customizer when it loads
2. Can't interact with the page "behind" the customizer

We can't fix the Add a Widget focus part in Calypso since it's in core, so I will close this issue in favor of the trac ticket: 50846-core#ticket
